core/metacling/src/libCling.script

```diff
@@ -0,0 +1,13 @@
+{
```

Comment:
The linker script approach used to make the library artifact much larger than without and using hidden visibility.

Reply:
Here I'm proposing to use the linker script in addition to all compilation flags that we already apply. In a `Release` build with GCC, this actually reduces the size of `libCling.so` by 822 kB (of 120 MB).

Reply:
Regarding the size: ROOT v6.26 (released in early 2022) had LLVM 9 and `libCling.so` was already 73 MB. For ROOT v6.28 (released in early 2023), we upgraded to LLVM 13 and `libCling.so` grew to 100 MB. So the answer is that it's LLVM growing in size.